odoo/http.py

```diff
@@ -1160,17 +1161,25 @@ def load_request_data(self):
                     pass
 
 
-def session_gc(session_store):
-    if random.random() < 0.001:
-        # we keep session one week
-        last_week = time.time() - 60*60*24*7
-        for fname in os.listdir(session_store.path):
-            path = os.path.join(session_store.path, fname)
-            try:
-                if os.path.getmtime(path) < last_week:
-                    os.unlink(path)
-            except OSError:
-                pass
+ODOO_DISABLE_SESSION_GC = str2bool(os.environ.get('ODOO_DISABLE_SESSION_GC', '0'))
+
+if ODOO_DISABLE_SESSION_GC:
+    # empty function, in case another module would be
+    # calling it out of setup_session()
+    def session_gc(session_store):
+        return
+else:
+    def session_gc(session_store):
+        if random.random() < 0.001:
+            # we keep session one week
+            last_week = time.time() - 60*60*24*7
+            for fname in os.listdir(session_store.path):
+                path = os.path.join(session_store.path, fname)
+                try:
+                    if os.path.getmtime(path) < last_week:
+                        os.unlink(path)
+                except OSError:
+                    pass
```

Comment:
to reduce versioning noise and indentation, @antonylesuisse suggests to re-define `session_gc()` based on the env variable, leaving the original function untouched. How about something like this (restoring the original code for the rest):

```py
ODOO_DISABLE_SESSION_GC = str2bool(os.environ.get('ODOO_DISABLE_SESSION_GC', '0'))

if ODOO_DISABLE_SESSION_GC:
    # empty function, in case another module would be
    # calling it outside of setup_session()
    session_gc = lambda s: None
```